pkg/clusteragent/autoscaling/workload/controller.go

```diff
@@ -212,6 +212,7 @@ func (c *Controller) syncPodAutoscaler(ctx context.Context, key, ns, name string
 		if podAutoscalerInternal.Deleted() || podAutoscalerInternal.Spec().Owner != datadoghqcommon.DatadogPodAutoscalerRemoteOwner {
 			log.Infof("Object %s not present in Kubernetes and flagged for deletion (remote) or owner == local, clearing internal store", key)
 			c.store.UnlockDelete(key, c.ID)
+			deletePodAutoscalerTelemetry(podAutoscalerInternal.Namespace(), podAutoscalerInternal.Spec().TargetRef.Name, podAutoscalerInternal.Name())
```

Comment:
To avoid trying to track that, you can setup a `TelemetryObserver` on the `Store` and that will make sure you don't miss any source of deletion (we align store content with metrics).

With this I think the telemetry will not be accurate on follower -> leader transitions. Thinking about it, it's probably the same with the `update` BTW.

Reply:
Would it be inaccurate as we'd be incorrectly clearing the metrics on transition? Would that be fixed by the next reconcile/recommendation update